glmnet/linear.py

```diff
@@ -147,6 +159,12 @@ def __init__(self, alpha=1, n_lambda=100, min_lambda_ratio=1e-4,
         self.max_features = max_features
         self.verbose = verbose
 
+        if any(self.lower_limits) > 0 if isinstance(self.lower_limits, np.ndarray) else self.lower_limits > 0:
+            raise ValueError("lower_limits must be non-positive")
+
+        if any(self.upper_limits) < 0 if isinstance(self.upper_limits, np.ndarray) else self.upper_limits < 0:
```

Comment:
The second parentheses around the `any` call need to be after the `0`, here and above.
```python
 any(self.lower_limits > 0)
```

Another suggestion: Cast the inputs with an `np.asarray` if they fail an `np.isscalar` check; that will allow lists or tuples to be valid inputs. There should also be a `ValueError` raised if there's an array input with the wrong number of elements.

Please also put these checks in `fit`, maybe in their own helper function. The scikit-learn API requires that `__init__` do nothing other than set the input parameters.